cli/running/watchdog_test.go

```diff
@@ -58,7 +58,7 @@ func cleanupTempDir(tempDir string) {
 }
 
 // createTestWatchdog creates an instance and a watchdog for the test.
-func createTestWatchdog(t *testing.T, restartable bool) *Watchdog {
+func createTestWatchdog(t *testing.T, restartable bool) (*Watchdog, *providerTestImpl) {
```

Comment:
Provider is present in watchdog why are you returning it additionally?

Reply:
removed commit